uiowa/uiowa
# CORS tweaks

Relates to https://github.com/uiowa/uiowa/issues/7689

So, the previous _very cool_ approach to CORS was to add the necessary headers for specific paths, keeping the wildwest at bay, however, there was early issues that hinted at cached responses and those responses not always triggering and providing those headers.

Backing away from that solution for now, we are now using Drupal core functionality that is maintained at the services.yml level and is actually included but not turned on in the default site (default.services.yml). When it comes to Drupal multisite, the idea is to copy that file and only include the bits that you need for a specific site. In this case, I copied, gutted and modified the services file to enable the cors functionality.

Knowing that most requests to the emergency website could come from any domain and request the application/json content type header that is how I defined the configuration. Other example configurations go above and beyond and allow more things but each additional thing starts to expose us to potential XSS attacks if already vulnerable and performance hits.

content-type - Indicates the media type of the request body (e.g., application/json for JSON data)
allowedOrigins as * means that any domain like (testiowa.com, uiowa.edu, brand.uiowa.edu) can make a request.
allowedMethods as GET and OPTIONS keeps it restricted from doing things like POST which we don't need at this time.
other things are false by default so until we have a use-case, let's keep those off.



# How to test

- Use a cors tester like https://cors-test.codehappy.dev/ to set the URL to https://emergency.dev.drupal.uiowa.edu/api/active and the Origin to something like https://brand.uiowa.edu or https://gis.sites.uiowa.edu or https://signage.uiowa.edu with the method as GET.
- Observe that https://admissions.stage.drupal.uiowa.edu/ is showing a hawk alert that comes from emergency.dev.drupal.uiowa.edu and that there are no CORS related console errors.
- Authentication issue in DEV seems to be unrelated but was an early concern. Worse case, we can `drush uli` for Public Safety to make changes as we remove the changes.
